### PR TITLE
vc-sm-cma tidy ups, and one for bcm2835_codec

### PR DESCRIPTION
**Pull request by 6by9**
Error handling fixes.

### CODE REVIEW

**Comment by 6by9** [comment]
I can't see any more this time, and I think I'll stop looking  :-)

**Comment by 6by9** [comment]
@pelwell I'm good with this one now.

File: drivers/staging/vc04_services/vc-sm-cma/vc_sm.c [hunk 1]
```diff
@@ -757,16 +748,17 @@ static int bcm2835_vc_sm_cma_remove(struct platform_device *pdev)
...
 	if (sm_inited) {
 		/* Remove all proc entries. */
-		//debugfs_remove_recursive(sm_state->dir_root);
+		debugfs_remove_recursive(sm_state->dir_root);
 
 		/* Stop the videocore shared memory service. */
 		vc_sm_cma_vchi_stop(&sm_state->sm_handle);
+	}
 
+	if (sm_state) {
```
Comment (JamesH65) [inline drivers/staging/vc04_services/vc-sm-cma/vc_sm.c:757]: Isn't this guaranteed to be non-null?

Comment (JamesH65) [reply drivers/staging/vc04_services/vc-sm-cma/vc_sm.c:757]: Hold on, might have misread the diff.

Comment (6by9) [reply drivers/staging/vc04_services/vc-sm-cma/vc_sm.c:757]: No, I think you're right, but I'm now rereading it again.

Comment (6by9) [reply drivers/staging/vc04_services/vc-sm-cma/vc_sm.c:757]: You're right.
Should the kzalloc in probe fail then we'll bail out and remove will never be called. There should therefore never be a path where remove gets called with sm_state non-NULL.

**Comment by 6by9** [comment]
Force pushed an update to address James' comment.

**Comment by 6by9** [comment]
Rebased to 4.19.29

**Pull request merged**